### PR TITLE
Add support for multiple taxes per order line item

### PR DESCRIPTION
**Pull request by mntzrr**
Currently, we only support one tax rate per order line. Once we found one, we abort. This also applies to shipping line. With this PR, we'll handle all the taxes instead.

Task: https://app.clickup.com/t/865d0fzd6

### CODE REVIEW

File: src/Cart/CartLineItem.php [hunk 1]
```diff
@@ -193,7 +197,9 @@ public function set_total_discount_tax_amount() {
 	 * @return void
 	 */
 	public function set_total_tax_amount() {
-		$this->total_tax_amount = apply_filters( $this->get_filter_name( 'total_tax_amount' ), $this->format_price( $this->cart_item['line_tax'] ), $this->cart_item );
+		$taxes                  = count( $this->cart_item['line_tax_data']['total'] );
+		$total_tax_amount       = $taxes > 1 ? array_sum( $this->cart_item['line_tax_data']['total'] ) : $this->cart_item['line_tax'];
```
Comment (MichaelBengtsson) [inline src/Cart/CartLineItem.php:201]: @mntzrr Does the line_tax not already have the total of the tax data?

File: src/Cart/CartLineItem.php [hunk 2]
```diff
@@ -135,14 +135,18 @@ public function set_subtotal_unit_price() {
...
+			$tax           = new \WC_Tax();
+			$tax_rates     = $tax->get_rates( $this->product->get_tax_class() );
+			$item_tax_rate = array_sum(
+				array_map(
+					function ( $i ) {
+						return round( $i['rate'] * 100 ) ?? 0;
+					},
+					$tax_rates
+				)
+			);
```
Comment (MichaelBengtsson) [inline src/Cart/CartLineItem.php:147]: Lets discuss this internally a bit. I'm not sure if combining them all together is the correct thing here, or if we should refactor this to be able to return multiple different tax rates instead as an array.

**Review by MichaelBengtsson — COMMENTED** [review]
Lets discuss this a bit internally before we merge this and release.

File: src/Cart/CartLineItem.php (hunk 1 above)
Comment (mntzrr) [reply src/Cart/CartLineItem.php]: You're correct. I reverted the changes.

**Review by MichaelBengtsson — APPROVED** [review]

**Pull request merged**